### ISSUE TITLE
Memory leak!!!!

### ISSUE THREAD

**Issue by ChahalSandeep** [swatbotics/apriltag#6]
Hello, I was playing around with a script for one of my project. Unfortunately, the program leaks memory and its exponential. I am not a C programmer. I could find the leak. 
Here is my python script-

``` python
import cv2
import apriltag
capt = cv2.VideoCapture("some_video_stream")
detec = apriltag.Detector() 
while capt.isOpened():
    ret, frame = capt.read()
    # converting image to gray scale
    gray = cv2.cvtColor(frame, `cv2.COLOR_BGR2GRAY)`
    # passing it to function in apriltag.py
    **detected_codes_in_image = detec.detect(gray)**
   #some display function 
   if cv2.waitKey(1) & 0xFF == ord('q'):
        break
```

I could trace it with **resource** package in python
code in apriltag.py where the issue occurs. its the 
**self.libc.apriltag_detector_detect(self.tag_detector, c_img)**

``` python

class Detector(object):   # this class
    def detect(self, img, return_image=False): # this function

        '''Run detectons on the provided image. The image must be a grayscale
image of type numpy.uint8.'''

        assert len(img.shape) == 2
        assert img.dtype == numpy.uint8

        c_img = self._convert_image(img)

        # this line increases memory after few iteration.
        detections = self.libc.apriltag_detector_detect(self.tag_detector, c_img)

        #create a pytags_info object
        return_info = []

        for i in range(0, detections.contents.size):

            #extract the data for each apriltag that was identified
            apriltag = ctypes.POINTER(_ApriltagDetection)()
            self.libc.zarray_get(detections, i, ctypes.byref(apriltag))

            tag = apriltag.contents

            homography = _matd_get_array(tag.H).copy()
            center = numpy.ctypeslib.as_array(tag.c, shape=(2,)).copy()
            corners = numpy.ctypeslib.as_array(tag.p, shape=(4, 2)).copy()

            detection = Detection(
                tag.family.contents.name,
                tag.id,
                tag.hamming,
                tag.goodness,
                tag.decision_margin,
                homography,
                center,
                corners)

            #Append this dict to the tag data array
            return_info.append(detection)

```

**Comment by mzucker** [comment]
Commit 776a3910ccd414973487a644ab4915d9fb7725d9 frees some memory that I forgot to free, can you please see if behavior persists? Thanks!

**Comment by ChahalSandeep01** [comment]
@mzucker aren't you missing setup file?

**Comment by mzucker** [comment]
What do you mean?

**Comment by ChahalSandeep01** [comment]
I mean before commit you had setp.py file.

**Comment by ChahalSandeep** [comment]
thank you @mzucker . Let's see if this fixes it.

**Comment by mzucker** [comment]
Ok, I'll take another look soon.

**Comment by mzucker** [comment]
Ok, I found a few more leaks and fixed them in commit 15db309 – I also discovered the https://pypi.org/project/memory_profiler/ module along with its tool mprof which helped. Please let me know if you still see increasing memory usage over time.

**Comment by ChahalSandeep** [comment]
Thank for the help @mzucker , I have got issue fixed in commit [49cbf41](https://github.com/ChahalSandeep/apriltag/commit/49cbf41d9b38994e40bb84c07e7556f8cea3c6ba)

**Comment by mzucker** [comment]
Great!
